src/Files.App/Shell/Win32API.cs

```diff
@@ -280,58 +278,61 @@ public static (byte[]? icon, byte[]? overlay) GetFileIconAndOverlay(string path,
                         <= 48 => Shell32.SHIL.SHIL_EXTRALARGE,
                         _ => Shell32.SHIL.SHIL_JUMBO,
                     };
-                    if (!Shell32.SHGetImageList(imageListSize, typeof(ComCtl32.CImageList).GUID, out var imageListOut).Succeeded)
-                        return (iconData, null);
+                    lock (_lock)
```

Comment:
I don't get why we will need a lock here. Is it mentioned by some API documents?

Reply:
I found nothing in the API's documentation.

I reintroduced the lock to avoid an exception that was sometimes thrown by that part of the code. 
Here is the exception message:
`"The COM object that was separated from its underlying RCW cannot be used. The COM object was released while it was still in use on another thread."`

I believe this happens when one thread executes line # 283 while another thread has already executed line # 334.